tests/tasks/setup/kitctl/dataplane.yaml

```diff
@@ -0,0 +1,52 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: data-plane-setup
+  namespace: tekton-pipelines
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/categories: Kubernetes
+    tekton.dev/tags: CLI, kubectl
+    tekton.dev/displayName: "kubernetes actions"
+    tekton.dev/platforms: "linux/amd64"
+spec:
+  description: |
+    Setup a kubernetes data plane in the guest cluster.
+  params:
+  - name: name
+    default: "guest"
+    description: Name of the guest cluster
+  - name: node-count
+    default: "5"
+    description: Number of worker nodes
+  steps:
+  - name: setup-data-plane
+    image: bitnami/kubectl
+    script: |
+      #!/bin/bash
+      echo "Getting subnet and security group tags"
+      TAG=$(kubectl get provisioner -oyaml | grep karpenter.sh/discovery | awk 'NR==1{ print $2}')
+      echo "Setting up data plane"
+      cat <<EOF | kubectl apply -f -
+      apiVersion: kit.k8s.sh/v1alpha1
+      kind: DataPlane
+      metadata:
+        name: $(params.name)
+        namespace: $(params.name)
+      spec:
+        clusterName: $(params.name) # Associated Cluster name
+        nodeCount: $(params.node-count)
+        subnetSelector:
+          karpenter.sh/discovery: ${TAG}
+      EOF
+      echo "Getting kube admin config"
+      kubectl get secret -n $(params.name) $(params.name)-kube-admin-config -ojsonpath='{.data.config}' | base64 -d > /tmp/kubeconfig
+      echo "Waiting for the worker nodes to be ready"
+      while true; do
+          ready_node=$(kubectl --kubeconfig=/tmp/kubeconfig get nodes 2>/dev/null | grep -w Ready | wc -l)
+          if [[ "$ready_node" -eq $(params.node-count) ]]; then break; fi
```

Comment:
does it ever be equal even when all the nodes are ready ? I think you may get one more than the actual nodecount with `wc -l`

Reply:
The `grep -w Ready` is filtering out the first row of the output so that only nodes that are `Ready` will be outputted.  `wc -l` will count the correct number of nodes.